### PR TITLE
Added more accurate details on final constant use

### PR DESCRIPTION
**Pull request by hkatzdev**
Previous explanations was lightly incorrect - I tried to fix it without over complicating the subject.

- [X] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `master` branch of freeCodeCamp.
- [X] None of my changes are plagiarized from another source without proper attribution.
- [X] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #XXXXX

### CODE REVIEW

**Comment by HarrisonLKatz** [comment]
Could you please retry the check? It only failed due to a bug that was fixed by #24240.

**Comment by HarrisonLKatz** [comment]
All checks have passed now - please let me know if there is anything else I need to do on this pull request. Thank you.

**Comment by HarrisonLKatz** [comment]
All checks have passed now - please let me know if there is anything else I need to do on this pull request. Thank you.

File: guide/english/java/final-keyword/index.md [hunk 1]
```diff
@@ -3,7 +3,7 @@ title: Final
...
+You use the `final` keyword to make a variable’s reference constant. A constant variable can only be declared and initialized, which means that the final variable’s reference can not be changed once set. A final variable also has special rules about initialization. If you assign a class variable as final, you can only initialize in the declaration statement and the static block, making it constant the throughout the whole program. If it is a instance variable, you can only initialize in the declaration statement, the instance initialization block, and the constructor, making it constant throughout the object. If it is a local variable, you may initialize it anywhere after it is declared but before the method ends, making it constant after it’s initialized for however long the method last. Lastly, if it is a parameter variable then the value that is passed in may not be changed, making it constant for however long the method lasts. The reference and value of a variable are different for objects, but the same for primitive types.
```
Comment (good-praxis) [inline guide/english/java/final-keyword/index.md]: You're actually using the wrong kind of apostrophes in this text. You are using `’`, the correct one should be `'`. This is a minor change, though.

**Review by good-praxis — CHANGES_REQUESTED** [review]
I _personally_ think the change makes the description too explicit, however, I think it's a valid PR

File: guide/english/java/final-keyword/index.md (hunk 1 above)
Comment (HarrisonLKatz) [reply guide/english/java/final-keyword/index.md]: If you would like, I can try to make it a bit shorter this weekend.

**Review by Manish-Giri — CHANGES_REQUESTED** [review]
Hi @HarrisonLKatz 

While your changes are valid, please consider adding formatting (ex. bullet points) to these statements, so that they're easier to read -
> If you assign a class variable as final, you can only initialize in the declaration statement and the static block, making it constant the throughout the whole program. If it is a instance variable, you can only initialize in the declaration statement, the instance initialization block, and the constructor, making it constant throughout the object. If it is a local variable, you may initialize it anywhere after it is declared but before the method ends, making it constant after it’s initialized for however long the method last. Lastly, if it is a parameter variable then the value that is passed in may not be changed, making it constant for however long the method lasts. The reference and value of a variable are different for objects, but the same for primitive types.

Like so -

- If you assign a class variable as final, you can only initialize in the declaration statement and the static block, making it constant the throughout the whole program. 
- If it is a instance variable, you can only initialize in the declaration statement, the instance initialization block, and the constructor, making it constant throughout the object. 
- If it is a local variable, you may initialize it anywhere after it is declared but before the method ends, making it constant after it’s initialized for however long the method last.

Feel free to review the [style guide for guide articles](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/docs/style-guide-for-guide-articles.md) if you have any questions about formatting your changes.

Once you make these changes, I will be happy to review and merge your PR.

**Review by Manish-Giri — COMMENTED** [review]
Hi,

As there has been no response from your end for a while, I am closing this PR.

Feel free to re-open it if you wish to apply the suggested changes.

Happy Coding!

**Comment by HarrisonLKatz** [comment]
Hello,
Sorry for no responses for a while, I have been very busy with school work unfortunately. I will submit a new PR in a few days - would you like me to reopen the PR now or once I have made the changes?
Thank you,
Harrison

**Comment by Manish-Giri** [comment]
Hi,

That's all right, take your time. I've re-opened the PR.

**Comment by Manish-Giri** [comment]
Closing the PR. Please re-open when you're ready to re-submit. 

Thank You.

**Pull request closed without merging**